src/main/kotlin/me/zeroeightsix/kami/feature/module/Notebot.kt

```diff
@@ -160,46 +130,43 @@ object Notebot : Module() {
             } else displayActionbar(player, "You are in creative mode and cannot play music.")
         }
-        lastNote = System.currentTimeMillis()
     })
 
     private fun playNotes(notes: List<Note>, player: ClientPlayerEntity, world: World) {
-        val blockPosArr: ArrayList<BlockPos?> = ArrayList()
+        val blockPosArr = mutableListOf<BlockPos?>()
         notes.forEach { n ->
             val channelsArray: Array<Instrument> =
                 arrayOf(channelZero, channelOne, channelTwo, channelThree, channelFour)
-            val number = (n.track % (channelsArray.size - 1))
-            val enum = channelsArray[if (number < 0) 0 else number]
-            displayActionbar(player, enum.toString())
-            blockPosArr.add(instrumentMap[enum][if (enum == Instrument.HAT && hatAlwaysAsFSharp) 0 else n.notebotNote])
+            val number = n.track % (channelsArray.size - 1)
+            val instrument = channelsArray[if (number < 0) 0 else number]
+            blockPosArr.add(instrumentMap[instrument][if (instrument == Instrument.HAT && hatAlwaysAsFSharp) 0 else n.notebotNote])
         }
         playBlock(blockPosArr, player, world)
     }
 
-    private fun playBlock(blockPosTracks: ArrayList<BlockPos?>, player: ClientPlayerEntity, world: World) {
-        if (blockPosTracks.isNotEmpty()) {
-            blockPosTracks.forEach { blockPos ->
-                if (blockPos != null) {
-                    getIrreplaceableNeighbour(world, blockPos)?.let { (_, direction) ->
-                        mc.networkHandler?.sendPacket(
-                            PlayerActionC2SPacket(PlayerActionC2SPacket.Action.START_DESTROY_BLOCK, blockPos, direction)
-                        )
-                        mc.networkHandler?.sendPacket(
-                            PlayerActionC2SPacket(PlayerActionC2SPacket.Action.ABORT_DESTROY_BLOCK, blockPos, direction)
-                        )
-                        player.swingHand(Hand.MAIN_HAND)
-
-                        mc.itemUseCooldown = 4
+    private fun playBlock(blockPosTracks: List<BlockPos?>, player: ClientPlayerEntity, world: World) {
+        blockPosTracks.asSequence().filterNotNull().forEach { blockPos ->
+            getIrreplaceableNeighbour(world, blockPos)?.let { (_, direction) ->
+                if (player.pos.isInRange(Vec3d(blockPos.x.toDouble(), blockPos.y.toDouble(), blockPos.z.toDouble()), 6.0)) {
+                    mc.networkHandler?.let {
+                        it.sendPacket(PlayerActionC2SPacket(PlayerActionC2SPacket.Action.START_DESTROY_BLOCK, blockPos, direction))
+                        it.sendPacket(PlayerActionC2SPacket(PlayerActionC2SPacket.Action.ABORT_DESTROY_BLOCK, blockPos, direction))
                     }
+                    player.swingHand(Hand.MAIN_HAND)
+                    mc.itemUseCooldown = 4
+                } else {
+                    displayActionbar(player, "You are not in range to play this block.")
```

Comment:
Which block? Give some coordinates.

You may also want to mention that, in order for notebot to recognise a new set of noteblocks, the user has to click the `Discover` button.